src/bucket/FutureBucket.cpp

```diff
@@ -253,9 +256,28 @@ FutureBucket::getOutputHash() const
     return mOutputBucketHash;
 }
 
+static double
+getAvailableTimeForMerge(uint32_t level)
+{
+    double availableTime = 5;
+    if (level == 1)
+    {
+        availableTime = 10;
+    }
+    else if (level > 1)
+    {
+        for (uint32_t i = 0; i < level; ++i)
```

Comment:
I'm not sure what this function is calculating: I _think_ the idea is to estimate how many seconds a future bucket has to perform a merge before it is needed? If so, the formula for that is `availableTime(level i)  = BucketList::levelHalf(level i - 1) * timeToCloseLedger` -- in other words, level i has to promote future bucket and start a new merge every half-size of level i-1. We also don't need to special case level 1, as there isn't anything different about it. I found this comment quite helpful in understanding the mechanics of level spilling: https://github.com/stellar/stellar-core/blob/dc5fab0e6b6ba505a615de521bbf1beff7fd0d62/src/bucket/BucketList.cpp#L475)